core/src/main/java/hudson/scm/SCM.java

```diff
@@ -522,13 +522,24 @@ public void postCheckout(AbstractBuild<?,?> build, Launcher launcher, FilePath w
      * (for example, SVN revision number.)
      *
      * <p>
-     * This method is invoked whenever someone does {@link AbstractBuild#getEnvironment(TaskListener)}, which
+     * This method is invoked whenever someone does {@link Run#getEnvironment(TaskListener)}, which
      * can be before/after your checkout method is invoked. So if you are going to provide information about
      * check out (like SVN revision number that was checked out), be prepared for the possibility that the
      * check out hasn't happened yet.
+     *
+     * @since FIXME
      */
-    // TODO is an equivalent for Run needed?
+    public void buildEnvironment(@Nonnull Run<?,?> build, @Nonnull Map<String,String> env) {
+        if (build instanceof AbstractBuild) {
+            buildEnvVars((AbstractBuild)build, env);
```

Comment:
Ah, right, was thinking of the case where the delegation is done in the opposite order.